db/view/view.cc

```diff
@@ -1452,6 +1493,13 @@ future<stop_iteration> view_update_builder::on_results() {
 
     auto tombstone = std::max(_update_partition_tombstone, _update_current_tombstone);
     if (tombstone && _existing && !_existing->is_end_of_partition()) {
+        if (_update_partition_tombstone && !_update_current_tombstone) {
+            // The remaining updates are only for the partition tombstone.
+            // For views that have the same partition key as base, generate an update of partition tombstone to delete
+            // the entire partition in one operation, instead of generating an update for each row.
+            generate_update(_update_partition_tombstone);
+        }
+
```

Comment:
It would be really great if this was called at most once during the view_builder's lifetime. This way we could get rid of the `view_updates::_update_done` variable. The best way IMO would be not to add more flags but rather put it in such a place / behind such conditions that it gets called at most once, however I don't have a particular suggestion on how to exactly do that right away.

For example, maybe we could do that right after we process the partition start fragments.

Reply:
I have moved this to build_some when we process the partition start, I agree it is more clear this way
I was still not able to get rid of _update_done because it applies to only the specific views that satisfy the same partition key condition
In general the view builder can have views that don't satisfy this so it still needs to go over all the rows